### PR TITLE
Improvement/reduce scrolling

### LINKED ISSUE #88: Reduce Amount of Scrolling
**Is your feature request related to a problem? Please describe.**
There is a lot of scrolling require to see everything currently.

**Describe the solution you'd like**
Reduce screen real-estate used and/or create another tab focused on visualizing at a wider view.

**Additional context**
See Pruim's original schedulizer for potential inspiration.

### PR DESCRIPTION
**Pull request by bryantgeorge**
This PR generally tries to make less scrolling required, to do so it:
1. Makes widths (and text sizes) of faculty/room schedules constant pixel sizes, so that in theory you should be able to fit 5 schedules instead of 4 on a page with a sufficiently wide monitor. Notice that when expanding the width of the browser window, the schedules stay the same width and more schedules are available.
2. Creates a `Department Schedule` tab which has the courses grouped by the faculty name (should be the same for all courses in the math schedule). These schedules are 4 times as wide, so that you can see the courses. This is intended to help dept chairs to see the entire schedule at a glance.

Things I would recommend testing:
1. Try on chrome, firefox, and safari (if possible).
2. You could try the department view for the `Course Section Enrollment.csv` which looks really cool to have all of the department schedules in a semi-efficient space. (In my experience, using this schedule can be really computationally intensive and switching from the `Faculty Schedule` tab to the `Department Schedule` tab will take a while.)

Closes #88

### CODE REVIEW

**Comment by charkour** [comment]
In #176, I split all of the Schedule.tsx components into their own files so there will be some conflicts with this PR. Once this PR is approved, we can marge this in and I'll worry about reconciling the issues! 

I'll check this out tomorrow likely! Thanks for the good work.

File: client-course-schedulizer/src/utilities/hooks/addSectionHooks.ts [hunk 1]
```diff
@@ -118,7 +118,7 @@ const scrollToUpdatedSection = (newCourse: Course, newSection: Section) => {
     });
   });
   if (newElement && newElement.parentElement?.parentElement) {
-    newElement.scrollIntoView({ behavior: "smooth", block: "nearest", inline: "nearest" });
+    newElement.scrollIntoView(false);
```
Comment (charkour) [inline client-course-schedulizer/src/utilities/hooks/addSectionHooks.ts:121]: Is this change on purpose?

**Review by charkour — CHANGES_REQUESTED** [review]
Looks good to me! Thanks for updating the styles and the dept tab looks really cool! If we had more time, we could look into memorization and virtualization to increase rendering performance, but tbh I don't understand either of those technologies. 

Nice work.

File: client-course-schedulizer/src/utilities/hooks/addSectionHooks.ts (hunk 1 above)
Comment (bryantgeorge) [reply client-course-schedulizer/src/utilities/hooks/addSectionHooks.ts:121]: Yes, I forgot that I did this, but this is to make it compatible with Safari, see:

https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView
image

Safari doesn't support the `scrollIntoViewOptions` parameter, but they do support the `alignToTop` boolean parameter. Unfortunately, this will cause the scroll into view to not be smooth, but I think it's worth it. (Also the scroll into view for the faculty schedule still has the `scrollIntoViewOptions` parameter, but I figure if that one doesn't work on Safari it isn't the biggest deal)

Comment (charkour) [reply client-course-schedulizer/src/utilities/hooks/addSectionHooks.ts:121]: Gotcha, thanks!
